### ISSUE TITLE
Colours on lines?

### ISSUE THREAD

**Issue by GeoffMurray-chc** [gamma-data/GCP-C4-PlantUML#3]
Is there a way to set a colour on a Rel object?

For example, this changes the colour of the container to red. Adding colours is a super handy visual for showing changes (new / changed / deleted / unchanged) in diagrams as you progress an implementation. 

ContainerOptimizedOS(AppConfig,"Application Configuration Service","Ipi.AppConfig","") #Red

Is there a way to do an equivalent on this statement:

Rel(CHRSServer,DAuth,"") 

I couldn't find a way to do so. 

I'm not suggesting it has to be as easy as it happens to work on the other elements, but is there a technique that can be used?

**Comment by jeffreyaven** [comment]
Hi @GeoffMurray-chc this is possible, let me look into it, did you just want the connections to be red for new relationships and black for existing relationships?

**Comment by GeoffMurray-chc** [comment]
The red was an example. The colours I'm currently using (but might want to be adjusted by my real user community :-), are:

Unchanged element - default colors of the item
Modified element - yellow (which may not work too well with lines)
New element - light green
Deleted element - orange (red was too harsh for a large object)

I actually have code that takes an excel listing, and generates the C4 UML (I'm trying to find ways to ensure people can easily maintain the set of objects and boundaries and quickly see errors), so from that perspective, the object coloring I'm using is easily changeable, and then we just regenerate the PlantUML code. 

If it makes more sense to do some kind of mapping defined somehow in the PlantUML, that'd be ok, too. 

Does that help clarify?

**Comment by jeffreyaven** [comment]
OK @GeoffMurray-chc makes sense let me look at the best way to implement this

**Comment by jeffreyaven** [comment]
Hi @GeoffMurray-chc I have added support for `_New`,` _Modified` and `_Deleted`, try the example at [https://github.com/gamma-data/GCP-C4-PlantUML/blob/master/examples/gcp-C4-example-ext.puml](https://github.com/gamma-data/GCP-C4-PlantUML/blob/master/examples/gcp-C4-example-ext.puml) let me know how you go with this

**Comment by GeoffMurray-chc** [comment]
Thank you for taking the time. I'm not quite following how to use it. The example looks like original statements to me, but I could be missing something.

**Comment by jeffreyaven** [comment]
hey @GeoffMurray-chc notice the `_New`,`_Modified` and `_Deleted` suffixes on the resources, e.g. `CloudStorage_Deleted(...` these suffixes will change the border colors accordingly...

**Comment by GeoffMurray-chc** [comment]
I'm so sorry - of course. Does it work for lines as well?

**Comment by jeffreyaven** [comment]
not at the moment @GeoffMurray-chc is this what you wanted?

**Comment by GeoffMurray-chc** [comment]
Yeah, that's what I was hoping for. I already found a workaround for the boundaries and the elements within (described in the original post with the addition of the colours at the end of the object). It was the lines I was really having trouble with.

**Comment by TomKlimovski** [comment]
that's awesome Geoff. Just out of curiosity, where are you working that makes this repo is of value to you? Jeff Aven and myself are always looking for new use-cases for this library, perhaps we can move a few of your use cases up in priority if you see value in it.

**Comment by GeoffMurray-chc** [comment]
I'm sort of early stages, but this is what I'm doing. We have very detailed Information Security reviews for our deployments. We need to be able to show current baseline, as well as changes. We also need to be able to focus on certain message flows, so that our various stakeholders understand what is happening under the hood. Our system (60+ services) is deployed using yaml files (combination of lots of GKE and also other Google services). I'm translating those yaml files into a simplified structure so that I can then do a couple of things:

- show the current "as-built" diagram - this is not the live information from Google, but the deployment yaml files I talked about above that were used to deploy - my plan, at the moment, is to build a C4 diagram based on that. 
- allow the teams to focus on particular message flows (ie, if they take a message flow defined, build a diagram consisting only of that message flow into the system (or out of the system, depending on where it does), using the same basic as built information, but only a subset (ensures the overall diagram is consistent, whether you look at a portion or whole thing).
- allow the users to mark some elements and connections as new / modified / deleted before they generate the diagrams
- allow the users to generate new elements that might not be in the deployment files yet (ie, early in the design), and generate diagrams consistent with the deployed version. 

I'm at the first stage (translating them). I've built an excel prototype version of some of the rest, which I want to translate into something a little easier to manage for the design team. I'm hoping this saves us time overall in terms of prepping the right material for the Infosec reviews and sharing the focused message flow information with other teams. The reason I'm poking at the colours is that it's very helpful for information context. 

Make sense?
